### PR TITLE
Passwordless SSH access for PostgreSQL used in cloning standby nodes

### PR DESCRIPTION
**Pull request by atsikham**

### CODE REVIEW

File: ansible/playbooks/roles/postgresql/tasks/helpers/configure-ssh-access.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      file:
+        path: "{{ home_dir }}/.ssh/"
+        state: directory
+        mode: u=rwx,go=
+
+    - name: Generate openssh keypair
+      openssh_keypair:
+        path: "{{ home_dir }}/.ssh/id_ed25519"
+        type: ed25519
+        regenerate: full_idempotence
```
Comment (plirglo) [inline ansible/playbooks/roles/postgresql/tasks/helpers/configure-ssh-access.yml:79]: Did you consider to create backup of existing key?

"If set to ```full_idempotence```, the key will be regenerated if it does not conform to the module's options. This is also the case if the key cannot be read (broken file), the key is protected by an unknown passphrase, or when they key is not protected by a passphrase, but a passphrase is specified. Make sure you have a ```backup``` when using this option!" [DOC](https://docs.ansible.com/ansible/latest/collections/community/crypto/openssh_keypair_module.html#parameter-regenerate)

**Review by erzetpe — DISMISSED** [review]

**Review by plirglo — DISMISSED** [review]

**Comment by przemyslavic** [comment]
Upgrade from epicli v1.0 (PostgreSQL v10) fails on Ubuntu:
```
2022-03-04T13:13:04.2860382Z[38;21m13:13:04 INFO cli.src.ansible.AnsibleCommand - TASK [postgresql : standby | extensions | repmgr | Clone standby node] *********
2022-03-04T13:13:04.8605006Z[33;21m13:13:04 WARNING cli.src.ansible.AnsibleCommand - [WARNING]: Using world-readable permissions for temporary files Ansible needs
2022-03-04T13:13:04.8609906Z[38;21m13:13:04 INFO cli.src.ansible.AnsibleCommand - to create when becoming an unprivileged user. This may be insecure. For
2022-03-04T13:13:04.8611071Z[38;21m13:13:04 INFO cli.src.ansible.AnsibleCommand - information on securing this, see https://docs.ansible.com/ansible-
2022-03-04T13:13:04.8612199Z[38;21m13:13:04 INFO cli.src.ansible.AnsibleCommand - core/2.12/user_guide/become.html#risks-of-becoming-an-unprivileged-user
2022-03-04T13:13:05.3353505Z[31;21m13:13:05 ERROR cli.src.ansible.AnsibleCommand - fatal: [ec2-63-33-xxx-xxx.eu-west-1.compute.amazonaws.com]: FAILED! => {"changed": false, "cmd": ["/usr/bin/repmgr", "standby", "clone", "-h", "ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com", "-U", "epi_repmgr_admin", "-d", "epi_repmgr", "-p", "5432", "--force", "--copy-external-config-files=samepath", "--remote-user=postgres"], "delta": "0:00:00.145678", "end": "2022-03-04 13:13:05.243278", "msg": "non-zero return code", "rc": 1, "start": "2022-03-04 13:13:05.097600", "stderr": "NOTICE: destination directory \"/var/lib/postgresql/13/main\" provided
INFO: connecting to source node
DETAIL: connection string is: host=ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com user=epi_repmgr_admin port=5432 dbname=epi_repmgr
DETAIL: current installation size is 31 MB
INFO: replication slot usage not requested;  no replication slot will be set up for this standby
NOTICE: checking for available walsenders on the source node (2 required)
NOTICE: checking replication connections can be made to the source server (2 required)
WARNING: data checksums are not enabled and \"wal_log_hints\" is \"off\"
DETAIL: pg_rewind requires \"wal_log_hints\" to be enabled
WARNING: unable to connect to remote host \"ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com\" via SSH
ERROR: remote host \"ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com\" is not reachable via SSH - unable to copy external configuration files", "stderr_lines": ["NOTICE: destination directory \"/var/lib/postgresql/13/main\" provided", "INFO: connecting to source node", "DETAIL: connection string is: host=ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com user=epi_repmgr_admin port=5432 dbname=epi_repmgr", "DETAIL: current installation size is 31 MB", "INFO: replication slot usage not requested;  no replication slot will be set up for this standby", "NOTICE: checking for available walsenders on the source node (2 required)", "NOTICE: checking replication connections can be made to the source server (2 required)", "WARNING: data checksums are not enabled and \"wal_log_hints\" is \"off\"", "DETAIL: pg_rewind requires \"wal_log_hints\" to be enabled", "WARNING: unable to connect to remote host \"ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com\" via SSH", "ERROR: remote host \"ec2-54-217-yyy-yyy.eu-west-1.compute.amazonaws.com\" is not reachable via SSH - unable to copy external configuration files"], "stdout": "", "stdout_lines": []}
```

There were no issues on RHEL:
```
2022-03-04T12:53:08.3620835Z [38;21m12:53:08 INFO cli.src.ansible.AnsibleCommand - TASK [postgresql : standby | extensions | repmgr | Clone standby node] *********[0m
2022-03-04T12:53:11.0149524Z [38;21m12:53:11 INFO cli.src.ansible.AnsibleCommand - changed: [ec2-15-236-zzz-zzz.eu-west-3.compute.amazonaws.com][0m
```

**Comment by przemyslavic** [comment]
I also encountered issues running multiple `repmgr standby switchover` commands on a 3-node (0, 1, 2) cluster:
```
[postgres@ci-devawsrhelflannel-postgresql-vm-0 ~]$ repmgr cluster show
 ID | Name                                 | Role    | Status    | Upstream                             | Location | Priority | Timeline | Connection string

----+--------------------------------------+---------+-----------+--------------------------------------+----------+----------+----------+---------------------------------------------------------------------
 1  | ci-devawsrhelflannel-postgresql-vm-0 | primary | * running |                                      | default  | 100      | 3        | host=10.1.6.173 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 2  | ci-devawsrhelflannel-postgresql-vm-1 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 2        | host=10.1.6.152 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 3  | ci-devawsrhelflannel-postgresql-vm-2 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 1        | host=10.1.6.75 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
```

Executed `repmgr standby switchover` on node1:
```
[postgres@ci-devawsrhelflannel-postgresql-vm-1 ~]$ repmgr cluster show
 ID | Name                                 | Role    | Status    | Upstream                             | Location | Priority | Timeline | Connection string

----+--------------------------------------+---------+-----------+--------------------------------------+----------+----------+----------+---------------------------------------------------------------------
 1  | ci-devawsrhelflannel-postgresql-vm-0 | standby |   running | ci-devawsrhelflannel-postgresql-vm-1 | default  | 100      | 1        | host=10.1.6.173 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 2  | ci-devawsrhelflannel-postgresql-vm-1 | primary | * running |                                      | default  | 100      | 2        | host=10.1.6.152 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 3  | ci-devawsrhelflannel-postgresql-vm-2 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 1        | host=10.1.6.75 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
```

Executed `repmgr standby switchover` on node0:
```
[postgres@ci-devawsrhelflannel-postgresql-vm-0 ~]$ repmgr cluster show
 ID | Name                                 | Role    | Status    | Upstream                             | Location | Priority | Timeline | Connection string

----+--------------------------------------+---------+-----------+--------------------------------------+----------+----------+----------+---------------------------------------------------------------------
 1  | ci-devawsrhelflannel-postgresql-vm-0 | primary | * running |                                      | default  | 100      | 3        | host=10.1.6.173 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 2  | ci-devawsrhelflannel-postgresql-vm-1 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 2        | host=10.1.6.152 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 3  | ci-devawsrhelflannel-postgresql-vm-2 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 1        | host=10.1.6.75 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
```

and then `repmgr standby switchover` on node2 and it failed on RHEL.
```
[postgres@ci-devawsrhelflannel-postgresql-vm-2 ~]$ repmgr standby switchover
NOTICE: executing switchover on node "ci-devawsrhelflannel-postgresql-vm-2" (ID: 3)
WARNING: no superuser connection available
DETAIL: it is recommended to perform switchover operations with a database superuser
HINT: provide the name of a superuser with -S/--superuser
WARNING: unable to connect to remote host "10.1.6.173" via SSH
ERROR: unable to connect via SSH to host "10.1.6.173", user ""
```

What's interesting, exactly the same scenario went smoothly on Ubuntu:
```
[postgres@ci-devawsubuflannel-postgresql-vm-2 ~]$ repmgr cluster show
 ID | Name                                | Role    | Status    | Upstream                            | Location | Priority | Timeline | Connection string

----+-------------------------------------+---------+-----------+-------------------------------------+----------+----------+----------+---------------------------------------------------------------------
 1  | ci-devawsubuflannel-postgresql-vm-0 | standby |   running | ci-devawsubuflannel-postgresql-vm-2 | default  | 100      | 3        | host=10.1.6.144 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 2  | ci-devawsubuflannel-postgresql-vm-1 | standby |   running | ci-devawsubuflannel-postgresql-vm-0 | default  | 100      | 2        | host=10.1.6.227 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 3  | ci-devawsubuflannel-postgresql-vm-2 | primary | * running |                                     | default  | 100      | 4        | host=10.1.6.238 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
```

**Comment by atsikham** [comment]
> I also encountered issues running multiple `repmgr standby switchover` commands on a 3-node (0, 1, 2) cluster:
> 
> ```
> [postgres@ci-devawsrhelflannel-postgresql-vm-0 ~]$ repmgr cluster show
>  ID | Name                                 | Role    | Status    | Upstream                             | Location | Priority | Timeline | Connection string
> 
> ----+--------------------------------------+---------+-----------+--------------------------------------+----------+----------+----------+---------------------------------------------------------------------
>  1  | ci-devawsrhelflannel-postgresql-vm-0 | primary | * running |                                      | default  | 100      | 3        | host=10.1.6.173 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  2  | ci-devawsrhelflannel-postgresql-vm-1 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 2        | host=10.1.6.152 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  3  | ci-devawsrhelflannel-postgresql-vm-2 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 1        | host=10.1.6.75 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
> ```
> 
> Executed `repmgr standby switchover` on node1:
> 
> ```
> [postgres@ci-devawsrhelflannel-postgresql-vm-1 ~]$ repmgr cluster show
>  ID | Name                                 | Role    | Status    | Upstream                             | Location | Priority | Timeline | Connection string
> 
> ----+--------------------------------------+---------+-----------+--------------------------------------+----------+----------+----------+---------------------------------------------------------------------
>  1  | ci-devawsrhelflannel-postgresql-vm-0 | standby |   running | ci-devawsrhelflannel-postgresql-vm-1 | default  | 100      | 1        | host=10.1.6.173 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  2  | ci-devawsrhelflannel-postgresql-vm-1 | primary | * running |                                      | default  | 100      | 2        | host=10.1.6.152 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  3  | ci-devawsrhelflannel-postgresql-vm-2 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 1        | host=10.1.6.75 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
> ```
> 
> Executed `repmgr standby switchover` on node0:
> 
> ```
> [postgres@ci-devawsrhelflannel-postgresql-vm-0 ~]$ repmgr cluster show
>  ID | Name                                 | Role    | Status    | Upstream                             | Location | Priority | Timeline | Connection string
> 
> ----+--------------------------------------+---------+-----------+--------------------------------------+----------+----------+----------+---------------------------------------------------------------------
>  1  | ci-devawsrhelflannel-postgresql-vm-0 | primary | * running |                                      | default  | 100      | 3        | host=10.1.6.173 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  2  | ci-devawsrhelflannel-postgresql-vm-1 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 2        | host=10.1.6.152 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  3  | ci-devawsrhelflannel-postgresql-vm-2 | standby |   running | ci-devawsrhelflannel-postgresql-vm-0 | default  | 100      | 1        | host=10.1.6.75 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
> ```
> 
> and then `repmgr standby switchover` on node2 and it failed on RHEL.
> 
> ```
> [postgres@ci-devawsrhelflannel-postgresql-vm-2 ~]$ repmgr standby switchover
> NOTICE: executing switchover on node "ci-devawsrhelflannel-postgresql-vm-2" (ID: 3)
> WARNING: no superuser connection available
> DETAIL: it is recommended to perform switchover operations with a database superuser
> HINT: provide the name of a superuser with -S/--superuser
> WARNING: unable to connect to remote host "10.1.6.173" via SSH
> ERROR: unable to connect via SSH to host "10.1.6.173", user ""
> ```
> 
> What's interesting, exactly the same scenario went smoothly on Ubuntu:
> 
> ```
> [postgres@ci-devawsubuflannel-postgresql-vm-2 ~]$ repmgr cluster show
>  ID | Name                                | Role    | Status    | Upstream                            | Location | Priority | Timeline | Connection string
> 
> ----+-------------------------------------+---------+-----------+-------------------------------------+----------+----------+----------+---------------------------------------------------------------------
>  1  | ci-devawsubuflannel-postgresql-vm-0 | standby |   running | ci-devawsubuflannel-postgresql-vm-2 | default  | 100      | 3        | host=10.1.6.144 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  2  | ci-devawsubuflannel-postgresql-vm-1 | standby |   running | ci-devawsubuflannel-postgresql-vm-0 | default  | 100      | 2        | host=10.1.6.227 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
>  3  | ci-devawsubuflannel-postgresql-vm-2 | primary | * running |                                     | default  | 100      | 4        | host=10.1.6.238 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
> ```

I can't reproduce that on RHEL, repeated multiple times:

```text
[postgres@atsikham-k8s-rhel-postgresql-vm-2 ~]$ repmgr standby switchover
NOTICE: executing switchover on node "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3)
WARNING: no superuser connection available
DETAIL: it is recommended to perform switchover operations with a database superuser
HINT: provide the name of a superuser with -S/--superuser
ERROR: local node "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3) is not a downstream of demotion candidate primary "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2)
DETAIL: registered upstream node ID is 1
HINT: execute "repmgr standby register --force" to update the local node's metadata

---

WARNING: following issues were detected
  - node "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3) is not attached to its upstream node "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2)

---

[postgres@atsikham-k8s-rhel-postgresql-vm-2 ~]$ repmgr standby follow --dry-run --upstream-node-id 2
INFO: follow target is primary node "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2)
INFO: replication connection to the local node was successful
INFO: replication connection to the follow target node was successful
INFO: local and follow target system identifiers match
DETAIL: system identifier is 7072310287278185886
INFO: timelines are same, this server is not ahead
DETAIL: local node lsn is 0/501C2B8, follow target lsn is 0/501C2B8
INFO: prerequisites for executing STANDBY FOLLOW are met

---

[postgres@atsikham-k8s-rhel-postgresql-vm-2 ~]$ repmgr standby follow --upstream-node-id 2
INFO: timelines are same, this server is not ahead
DETAIL: local node lsn is 0/501CDE0, follow target lsn is 0/501CDE0
NOTICE: setting node 3's upstream to node 2
WARNING: node "atsikham-k8s-rhel-postgresql-vm-2" not found in "pg_stat_replication"
NOTICE: STANDBY FOLLOW successful
DETAIL: standby attached to upstream node "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2)

---

[postgres@atsikham-k8s-rhel-postgresql-vm-2 ~]$ repmgr cluster show
 ID | Name                              | Role    | Status    | Upstream                          | Location | Priority | Timeline | Connection string                                                  
----+-----------------------------------+---------+-----------+-----------------------------------+----------+----------+----------+---------------------------------------------------------------------
 1  | atsikham-k8s-rhel-postgresql-vm-0 | standby |   running | atsikham-k8s-rhel-postgresql-vm-1 | default  | 100      | 2        | host=10.1.8.74 user=epi_repmgr dbname=epi_repmgr connect_timeout=2 
 2  | atsikham-k8s-rhel-postgresql-vm-1 | primary | * running |                                   | default  | 100      | 2        | host=10.1.8.227 user=epi_repmgr dbname=epi_repmgr connect_timeout=2
 3  | atsikham-k8s-rhel-postgresql-vm-2 | standby |   running | atsikham-k8s-rhel-postgresql-vm-1 | default  | 100      | 2        | host=10.1.8.124 user=epi_repmgr dbname=epi_repmgr connect_timeout=2   
 
 ---
 
 [postgres@atsikham-k8s-rhel-postgresql-vm-2 ~]$ repmgr standby switchover
NOTICE: executing switchover on node "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3)
WARNING: no superuser connection available
DETAIL: it is recommended to perform switchover operations with a database superuser
HINT: provide the name of a superuser with -S/--superuser
WARNING: 1 sibling nodes found, but option "--siblings-follow" not specified
DETAIL: these nodes will remain attached to the current primary:
  atsikham-k8s-rhel-postgresql-vm-0 (node ID: 1)
NOTICE: local node "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3) will be promoted to primary; current primary "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2) will be demoted to standby
NOTICE: stopping current primary node "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2)
WARNING: a superuser connection is required to issue a CHECKPOINT
HINT: provide a superuser with -S/--superuser
DETAIL: executing server command "sudo /bin/systemctl stop postgresql-13"
INFO: checking for primary shutdown; 1 of 60 attempts ("shutdown_check_timeout")
NOTICE: current primary has been cleanly shut down at location 0/6000028
INFO: user "epi_repmgr" does not have permission to execute "pg_promote()", falling back to "pg_ctl promote"
NOTICE: promoting standby to primary
DETAIL: promoting server "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3) using "/usr/pgsql-13/bin/pg_ctl  -w -D '/var/lib/pgsql/13/data' promote"
waiting for server to promote.... done
server promoted
NOTICE: waiting up to 60 seconds (parameter "promote_check_timeout") for promotion to complete
NOTICE: STANDBY PROMOTE successful
DETAIL: server "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3) was successfully promoted to primary
INFO: local node 2 can attach to rejoin target node 3
DETAIL: local node's recovery point: 0/6000028; rejoin target node's fork point: 0/60000A0
NOTICE: setting node 2's upstream to node 3
WARNING: unable to ping "host=10.1.8.227 user=epi_repmgr dbname=epi_repmgr connect_timeout=2"
DETAIL: PQping() returned "PQPING_NO_RESPONSE"
NOTICE: starting server using "sudo /bin/systemctl start postgresql-13"
WARNING: node "atsikham-k8s-rhel-postgresql-vm-1" not found in "pg_stat_replication"
NOTICE: NODE REJOIN successful
DETAIL: node 2 is now attached to node 3
NOTICE: node  "atsikham-k8s-rhel-postgresql-vm-2" (ID: 3) promoted to primary, node "atsikham-k8s-rhel-postgresql-vm-1" (ID: 2) demoted to standby
NOTICE: switchover was successful
DETAIL: node "atsikham-k8s-rhel-postgresql-vm-2" is now primary and node "atsikham-k8s-rhel-postgresql-vm-1" is attached as standby
NOTICE: STANDBY SWITCHOVER has completed successfully
```

**Review by erzetpe — DISMISSED** [review]

**Review by seriva — DISMISSED** [review]

**Comment by przemyslavic** [comment]
@atsikham I was able to easily reproduce the issue:
```
[postgres@ci-devawsrhelflannel-postgresql-vm-2 ~]$ repmgr standby switchover
NOTICE: executing switchover on node "ci-devawsrhelflannel-postgresql-vm-2" (ID: 3)
WARNING: no superuser connection available
DETAIL: it is recommended to perform switchover operations with a database superuser
HINT: provide the name of a superuser with -S/--superuser
WARNING: unable to connect to remote host "10.1.6.149" via SSH
ERROR: unable to connect via SSH to host "10.1.6.149", user ""
```

and Upgrade from v1.0 (Ubuntu) still fails:
```
14:37:58 INFO cli.src.ansible.AnsibleCommand - TASK [postgresql : standby | extensions | repmgr | Clone standby node] *********
14:37:59 WARNING cli.src.ansible.AnsibleCommand - [WARNING]: Using world-readable permissions for temporary files Ansible needs
14:37:59 INFO cli.src.ansible.AnsibleCommand - to create when becoming an unprivileged user. This may be insecure. For
14:37:59 INFO cli.src.ansible.AnsibleCommand - information on securing this, see https://docs.ansible.com/ansible-
14:37:59 INFO cli.src.ansible.AnsibleCommand - core/2.12/user_guide/become.html#risks-of-becoming-an-unprivileged-user
14:37:59 ERROR cli.src.ansible.AnsibleCommand - fatal: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]: FAILED! => {"changed": false, "cmd": ["/usr/bin/repmgr", "standby", "clone", "-h", "ec2-2-2-2-2.eu-west-1.compute.amazonaws.com", "-U", "epi_repmgr_admin", "-d", "epi_repmgr", "-p", "5432", "--force", "--copy-external-config-files=samepath", "--remote-user=postgres"], "delta": "0:00:00.120961", "end": "2022-03-14 14:37:59.694593", "msg": "non-zero return code", "rc": 1, "start": "2022-03-14 14:37:59.573632", "stderr": "NOTICE: destination directory \"/var/lib/postgresql/13/main\" provided\nINFO: connecting to source node\nDETAIL: connection string is: host=ec2-2-2-2-2.eu-west-1.compute.amazonaws.com user=epi_repmgr_admin port=5432 dbname=epi_repmgr\nDETAIL: current installation size is 31 MB\nINFO: replication slot usage not requested;  no replication slot will be set up for this standby\nNOTICE: checking for available walsenders on the source node (2 required)\nNOTICE: checking replication connections can be made to the source server (2 required)\nWARNING: data checksums are not enabled and \"wal_log_hints\" is \"off\"\nDETAIL: pg_rewind requires \"wal_log_hints\" to be enabled\nWARNING: unable to connect to remote host \"ec2-2-2-2-2.eu-west-1.compute.amazonaws.com\" via SSH\nERROR: remote host \"ec2-2-2-2-2.eu-west-1.compute.amazonaws.com\" is not reachable via SSH - unable to copy external configuration files", "stderr_lines": ["NOTICE: destination directory \"/var/lib/postgresql/13/main\" provided", "INFO: connecting to source node", "DETAIL: connection string is: host=ec2-2-2-2-2.eu-west-1.compute.amazonaws.com user=epi_repmgr_admin port=5432 dbname=epi_repmgr", "DETAIL: current installation size is 31 MB", "INFO: replication slot usage not requested;  no replication slot will be set up for this standby", "NOTICE: checking for available walsenders on the source node (2 required)", "NOTICE: checking replication connections can be made to the source server (2 required)", "WARNING: data checksums are not enabled and \"wal_log_hints\" is \"off\"", "DETAIL: pg_rewind requires \"wal_log_hints\" to be enabled", "WARNING: unable to connect to remote host \"ec2-2-2-2-2.eu-west-1.compute.amazonaws.com\" via SSH", "ERROR: remote host \"ec2-2-2-2-2.eu-west-1.compute.amazonaws.com\" is not reachable via SSH - unable to copy external configuration files"], "stdout": "", "stdout_lines": []}
```

**Review by przemyslavic — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**